xwiki-platform-core/xwiki-platform-web/xwiki-platform-web-templates/src/test/java/org/xwiki/web/DocumentTagsTest.java

```diff
@@ -90,7 +90,7 @@ void displayTagsWhenNoEditRightsAndTagPluginAvailableAndNoTags() throws Exceptio
         // - The tag label is displayed
         // - No tag is listed after the tag label
         // - No "+" link is displayed since the user doesn't have edit rights
-        assertThat(result, matchesPattern("\\Q<div class=\"doc-tags\" id=\"xdocTags\"> core.tags.list.label </div>"));
+        assertThat(result, matchesPattern("<div class=\"doc-tags\" id=\"xdocTags\"> </div>"));
```

Comment:
In my opinion, the reason for using ```matchesPattern``` instead of ```containsString``` is to check the exact format of the string. ```containsString``` only checks if a substring is present in the result, but it doesn't verify if it's in the correct format or if there are no extra or missing spaces/characters. By using a pattern, you can check for exact string format and avoid false negatives due to differences in formatting.

Reply:
Okay, I think you're right that we're in fact matching the whole string here, not a substring. But in this case it is even easier: Just replace it by an `assertEquals`
```suggestion
        assertEquals("<div class=\"doc-tags\" id=\"xdocTags\"> </div>", result);
```

You'll also need to add an appropriate import:
```java
import static org.junit.jupiter.api.Assertions.assertEquals;
```

There are other places where the pattern makes sense as it contains a match for an appropriate URL for example, but I think here it's just useless.

Reply:
Applied ```assertEquals``` :)